src/compiler/checker.ts

```diff
@@ -10547,25 +10584,18 @@ namespace ts {
                     // Skip past any prologue directives to find the first statement
                     // to ensure that it was a super call.
                     if (superCallShouldBeFirst) {
-                        let statements = (<Block>node.body).statements;
-                        let superCallStatement: ExpressionStatement;
-                        for (let statement of statements) {
-                            if (statement.kind === SyntaxKind.ExpressionStatement && isSuperCallExpression((<ExpressionStatement>statement).expression)) {
-                                superCallStatement = <ExpressionStatement>statement;
-                                break;
-                            }
-                            if (!isPrologueDirective(statement)) {
-                                break;
-                            }
+                        if (!isSuperCallFirstStatment) {
+                            error(superCallStatement, Diagnostics.A_super_call_must_be_the_first_statement_in_the_constructor_when_a_class_contains_initialized_properties_or_has_parameter_properties);

```

Comment:
The check is required to have a correct emit code. This error is used in the following case:
```typescript
class B {}
class C extends B {
    public x = 10;  // without this we won't give an error
    constructor() {
        var x = 10;
        super();
    }
}
```